metric/distribution/regular/regular_distribution.go

```diff
@@ -69,34 +69,33 @@ func (regularDist *RegularDistribution) Size() int {
 
 // weight is 1/samplingRate
 func (regularDist *RegularDistribution) AddEntryWithUnit(value float64, weight float64, unit string) error {
-	if weight > 0 {
-		if value < 0 {
-			return errors.New("negative value")
-		}
-		//sample count
-		regularDist.sampleCount += weight
-		//sum
-		regularDist.sum += value * weight
-		//min
-		if value < regularDist.minimum {
-			regularDist.minimum = value
-		}
-		//max
-		if value > regularDist.maximum {
-			regularDist.maximum = value
-		}
+	if weight <= 0 {
+		return fmt.Errorf("weight must be larger than 0: %v", weight)
```

Comment:
You sure this should change from debug to error (or I guess warning where it's logged at the caller)?  Is there any case where it could be too much spam?

Reply:
That's fair. I think it's more useful to return it as an error because then we can log the metric that caused it as well. Can change that to a debug log.

Reply:
Or have it so it logs the weight one as a debug and the value one as a warning.